galera/src/replicator_smm.cpp

```diff
@@ -259,6 +259,11 @@ galera::ReplicatorSMM::ReplicatorSMM(const struct wsrep_init_args* args)
     cert_.assign_initial_position(seqno, trx_proto_ver());
 
     build_stats_vars(wsrep_stats_);
+
+    // Now we have store directory name to conf. This directory name
+    // could later be used by other components, for example by gcomm
+    // to find appropriate location for view state file.
+    config_.set(BASE_DIR, data_dir_);
```

Comment:
I think it may be a tad too late under certain circumstances. I'd rather bring it closer to config object initialization. E.g.
```
    ...
    init_config_ (config_, args->node_address, args->data_dir),
    ...
    data_dir_ (config_.get(BASE_DIR)),
    ...
```